### PR TITLE
Refactor time range generation code

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

**Comment by daniel-thom** [comment]
All errors reported by mypy are in the time-mapping code, and so I'm assuming that @lixiliu will address them after we merge this to her branch.

File: src/chronify/sqlalchemy/functions.py [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
-from chronify.time_configs import DatetimeRange
+from chronify.time_configs import DatetimeRange, TimeBaseModel
+
+# Copied from Polars, which doesn't export the type.
+DbWriteMode: TypeAlias = Literal["replace", "append", "fail"]
 
 
 def read_database(
-    query: Selectable | str, conn: Connection, config: DatetimeRange
+    query: Selectable | str, conn: Connection, config: TimeBaseModel
```
Comment (lixiliu) [inline src/chronify/sqlalchemy/functions.py:15]: This is what I had before, but we only accept DatetimeRange though...

Comment (daniel-thom) [reply src/chronify/sqlalchemy/functions.py:15]: Not anymore. We have to be able to read a table that was registered in representative time.

**Review by lixiliu — APPROVED** [review]
LGTM, thanks Dan!

**Pull request merged**